core/src/main/java/io/questdb/griffin/engine/groupby/FillRangeRecordCursorFactory.java

```diff
@@ -94,6 +98,44 @@ public FillRangeRecordCursorFactory(
         this.samplingIntervalUnit = samplingIntervalUnit;
         this.timestampIndex = timestampIndex;
         this.valueFuncs = fillValues;
+        this.valueFuncsPos = fillValuesPos;
+
+        // only do this for value fill
+        if (!(valueFuncs.size() == 1 && valueFuncs.get(0).isNullConstant())) {
+            if (metadata.getColumnCount() - 1 > valueFuncs.size()) {
+                throw SqlException.$(fillValuesPos.getLast(), "not enough fill values");
+            }
+
+            /*
+            This is used to offset our lookup into the columns.
+            We don't expect the timestamp column to be included in the args list.
+            Therefore, once we pass the column, we need to offset back one entry to find
+            the corresponding fill value.
+             */
+            int passedTimestamp = 0;
```

Comment:
logic with skipping timestamp like this is flawed when columns are not arranged sequentially, .e.g. this fails with array index out of bounds:

```sql
create table y as (select timestamp_sequence(1000000, 100000) ts, rnd_double() t from long_sequence(1000000)) timestamp(ts) partition by day;

select first(t), ts from y
sample by 15m
fill(2.0);

```